src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -112,6 +113,25 @@ public string NamedUser
             }
         }
 
+        private InboxHandler onMessageCenterUpdated;
+        public event InboxHandler OnMessageCenterUpdated
+        {
+            add
+            {
+                onMessageCenterUpdated += value;
+                MessageCenterClass.Shared().Inbox.AddListener(this);
```

Comment:
If you call add more than once you will add the listener twice, causing duplicate events. You should only adad the listener once.